Add files via upload

antonychiu2 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

antonychiu2 committed:
Add files via upload

Commit:
Auto commit for SQL_Injection-1227713c-e393-432c-b06a-91e886610a28

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/1227713c-e393-432c-b06a-91e886610a28)

Commit:
Auto commit for SQL_Injection-662034c6-3ca8-4b51-9c56-7e9f494918c5

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/662034c6-3ca8-4b51-9c56-7e9f494918c5)

Commit:
Auto commit for SQL_Injection-8f7d5037-45a1-4daf-8acf-cc7ceb5ceeef

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/8f7d5037-45a1-4daf-8acf-cc7ceb5ceeef)

Commit:
Auto commit for SQL_Injection-d63d5d7c-6e3e-4598-9ed9-badd89dc5349

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/d63d5d7c-6e3e-4598-9ed9-badd89dc5349)

Commit:
Auto commit for SQL_Injection-8c02374e-64f0-4453-b367-367a72c11663

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/8c02374e-64f0-4453-b367-367a72c11663)

Commit:
Auto commit for SQL_Injection-21f94671-6d18-4db8-bdd9-38358a24ded3

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/21f94671-6d18-4db8-bdd9-38358a24ded3)

Commit:
Auto commit for SQL_Injection-961a883c-1353-45c6-b653-3fe153031cde

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/961a883c-1353-45c6-b653-3fe153031cde)

Commit:
Auto commit for SQL_Injection-8fefe99d-c02d-4747-b3e9-cc689063b67d

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/8fefe99d-c02d-4747-b3e9-cc689063b67d)

Commit:
Auto commit for SQL_Injection-0a2a4204-6a85-453c-9704-74c989ed090d

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/0a2a4204-6a85-453c-9704-74c989ed090d)

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_links.js b/web_server/config/models/graphs_links.js
--- a/web_server/config/models/graphs_links.js
+++ b/web_server/config/models/graphs_links.js
@@ -34,7 +34,7 @@
     getGraphLinksByZone: function (zone) {
         let limitQuery = { 'links': 1 };
         return graphLinksModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
 };
 
```

```diff
diff --git a/web_server/config/models/graphs_links.js b/web_server/config/models/graphs_links.js
--- a/web_server/config/models/graphs_links.js
+++ b/web_server/config/models/graphs_links.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/662034c6-3ca8-4b51-9c56-7e9f494918c5)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_docs.js b/web_server/config/models/graphs_docs.js
--- a/web_server/config/models/graphs_docs.js
+++ b/web_server/config/models/graphs_docs.js
@@ -31,7 +31,7 @@
     getGraphDocsByZone: function (zone) {
         let limitQuery = { 'docs': 1 };
         return graphDocsModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
 };
 
```

```diff
diff --git a/web_server/config/models/graphs_docs.js b/web_server/config/models/graphs_docs.js
--- a/web_server/config/models/graphs_docs.js
+++ b/web_server/config/models/graphs_docs.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/d63d5d7c-6e3e-4598-9ed9-badd89dc5349)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/cert_graphs.js b/web_server/config/models/cert_graphs.js
--- a/web_server/config/models/cert_graphs.js
+++ b/web_server/config/models/cert_graphs.js
@@ -31,6 +31,6 @@
 module.exports = {
     CertGraphModel: certGraphModel,
     getGraphDataByZone: function (zone) {
-        return certGraphModel.findOne({ 'zone': zone }).exec();
+        return certGraphModel.findOne({ 'zone': mongoSanitize.sanitize({ data: zone }).data }).exec();
     },
 };
 
```

```diff
diff --git a/web_server/config/models/cert_graphs.js b/web_server/config/models/cert_graphs.js
--- a/web_server/config/models/cert_graphs.js
+++ b/web_server/config/models/cert_graphs.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/0a2a4204-6a85-453c-9704-74c989ed090d)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -38,6 +38,6 @@
         }, limitQuery).exec();
     },
     getGraphCountByZone: function (zone) {
-        return (graphDataModel.countDocuments({ 'zone': zone }).exec());
+        return (graphDataModel.countDocuments({ 'zone': mongoSanitize.sanitize({ data: zone }).data }).exec());
     },
 };
 
```

```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/8f7d5037-45a1-4daf-8acf-cc7ceb5ceeef)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs.js b/web_server/config/models/graphs.js
--- a/web_server/config/models/graphs.js
+++ b/web_server/config/models/graphs.js
@@ -34,7 +34,7 @@
     getGraphConfigByZone: function (zone) {
         let limitQuery = { 'config': 1 };
         return graphModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
     getGraphCountByZone: function (zone) {
 
```

```diff
diff --git a/web_server/config/models/graphs.js b/web_server/config/models/graphs.js
--- a/web_server/config/models/graphs.js
+++ b/web_server/config/models/graphs.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/1227713c-e393-432c-b06a-91e886610a28)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -34,7 +34,7 @@
     getGraphDataByZone: function (zone) {
         let limitQuery = { 'data': 1, 'errs': 1 };
         return graphDataModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
     getGraphCountByZone: function (zone) {
 
```

```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/961a883c-1353-45c6-b653-3fe153031cde)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -45,7 +45,7 @@
         return (promise);
     },
     getTPDsByTPD: function (tpd) {
-        return tpdModel.findOne({ 'tld': tpd });
+        return tpdModel.findOne({ 'tld': mongoSanitize.sanitize({ data: tpd }).data });
     },
     getTPDsByWildcard: function (search, listOnly) {
         let promise;
 
```

```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/21f94671-6d18-4db8-bdd9-38358a24ded3)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -52,7 +52,7 @@
         let AWSregex = new RegExp('.*' + search + '$');
         if (listOnly) {
             promise = tpdModel.find({
-                'tld': { '$regex': AWSregex },
+                'tld': mongoSanitize.sanitize({ data: { '$regex': AWSregex } }).data,
             }, { 'tld': 1, 'zones.zone': 1 }).exec();
         } else {
             promise = tpdModel.find({ 'tld': { '$regex': AWSregex } }).exec();
 
```

```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/8fefe99d-c02d-4747-b3e9-cc689063b67d)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -36,7 +36,7 @@
     getTPDsByZone: function (zone, listOnly) {
         let promise;
         if (!listOnly) {
-            promise = tpdModel.find({ 'zones.zone': zone }).exec();
+            promise = tpdModel.find({ 'zones.zone': mongoSanitize.sanitize({ data: zone }).data }).exec();
         } else {
             promise = tpdModel.find({
                 'zones.zone': zone,
 
```

```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/9764df1d-444e-4e35-a475-9d591bd3edda/fix/8c02374e-64f0-4453-b367-367a72c11663)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Commit:
Auto commit for SQL_Injection-812e858f-7c75-44bd-bb86-b13c3a945364

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d1b8c29e-e288-4d80-8939-ad3a0f321d15/fix/812e858f-7c75-44bd-bb86-b13c3a945364)

Commit:
Auto commit for SQL_Injection-3ef7c217-6bd2-4d9b-8efe-45a6eca2e54a

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d1b8c29e-e288-4d80-8939-ad3a0f321d15/fix/3ef7c217-6bd2-4d9b-8efe-45a6eca2e54a)

Commit:
Auto commit for SQL_Injection-66c7b863-2740-4223-ae89-4b8fc564d9ec

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d1b8c29e-e288-4d80-8939-ad3a0f321d15/fix/66c7b863-2740-4223-ae89-4b8fc564d9ec)

Commit:
Auto commit for SQL_Injection-6f0a4db3-869d-4e01-99b2-5fc3dca09eaa

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d1b8c29e-e288-4d80-8939-ad3a0f321d15/fix/6f0a4db3-869d-4e01-99b2-5fc3dca09eaa)

Commit:
Auto commit for SQL_Injection-7745dadf-ae6f-4535-9dcf-ddd44f0ba6fa

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d1b8c29e-e288-4d80-8939-ad3a0f321d15/fix/7745dadf-ae6f-4535-9dcf-ddd44f0ba6fa)

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_docs.js b/web_server/config/models/graphs_docs.js
--- a/web_server/config/models/graphs_docs.js
+++ b/web_server/config/models/graphs_docs.js
@@ -30,7 +30,7 @@
 module.exports = {
     GraphDocsModel: graphDocsModel,
     getGraphDocsByZone: function (zone) {
-        let limitQuery = { 'docs': 1 };
+        let limitQuery = { 'docs': mongoSanitize.sanitize({ data: 1 }).data };
         return graphDocsModel.findOne({
             'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d1b8c29e-e288-4d80-8939-ad3a0f321d15/fix/6f0a4db3-869d-4e01-99b2-5fc3dca09eaa)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_links.js b/web_server/config/models/graphs_links.js
--- a/web_server/config/models/graphs_links.js
+++ b/web_server/config/models/graphs_links.js
@@ -33,7 +33,7 @@
 module.exports = {
     GraphLinksModel: graphLinksModel,
     getGraphLinksByZone: function (zone) {
-        let limitQuery = { 'links': 1 };
+        let limitQuery = { 'links': mongoSanitize.sanitize({ data: 1 }).data };
         return graphLinksModel.findOne({
             'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d1b8c29e-e288-4d80-8939-ad3a0f321d15/fix/3ef7c217-6bd2-4d9b-8efe-45a6eca2e54a)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -56,7 +56,7 @@
                 'tld': mongoSanitize.sanitize({ data: { '$regex': AWSregex } }).data,
             }, { 'tld': 1, 'zones.zone': 1 }).exec();
         } else {
-            promise = tpdModel.find({ 'tld': { '$regex': AWSregex } }).exec();
+            promise = tpdModel.find({ 'tld': mongoSanitize.sanitize({ data: { '$regex': AWSregex } }).data }).exec();
         }
         return (promise);
     },
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d1b8c29e-e288-4d80-8939-ad3a0f321d15/fix/7745dadf-ae6f-4535-9dcf-ddd44f0ba6fa)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs.js b/web_server/config/models/graphs.js
--- a/web_server/config/models/graphs.js
+++ b/web_server/config/models/graphs.js
@@ -33,7 +33,7 @@
 module.exports = {
     GraphModel: graphModel,
     getGraphConfigByZone: function (zone) {
-        let limitQuery = { 'config': 1 };
+        let limitQuery = { 'config': mongoSanitize.sanitize({ data: 1 }).data };
         return graphModel.findOne({
             'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d1b8c29e-e288-4d80-8939-ad3a0f321d15/fix/812e858f-7c75-44bd-bb86-b13c3a945364)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -40,7 +40,7 @@
             promise = tpdModel.find({ 'zones.zone': mongoSanitize.sanitize({ data: zone }).data }).exec();
         } else {
             promise = tpdModel.find({
-                'zones.zone': zone,
+                'zones.zone': mongoSanitize.sanitize({ data: zone }).data,
             }, { 'tld': 1, 'zones.zone': 1 }).exec();
         }
         return (promise);
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d1b8c29e-e288-4d80-8939-ad3a0f321d15/fix/66c7b863-2740-4223-ae89-4b8fc564d9ec)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.
